modules/pam_pwhistory/pwhistory_config.c

```diff
@@ -67,6 +72,19 @@ parse_config_file(pam_handle_t *pamh, int argc, const char **argv,
         fname = PWHISTORY_DEFAULT_CONF;
     }
 
+#ifdef VENDOR_PWHISTORY_DEFAULT_CONF
+    if (!strcmp(fname, PWHISTORY_DEFAULT_CONF)) {
+        /*
+         * Check whether PWHISTORY_DEFAULT_CONF file is available.
+         * If it does not exist, fall back to VENDOR_PWHISTORY_DEFAULT_CONF file.
+         */
+        struct stat buffer;
+        if (stat(fname, &buffer) != 0 && errno == ENOENT) {
+            fname = VENDOR_PWHISTORY_DEFAULT_CONF;
```

Comment:
I think we shouldn't fall back to the vendor config file if a `conf=` option was specified and its argument was equal to the default config file name.
That is, I suggest moving the fall back code inside the `fname == NULL` branch, this would also make the `strcmp` check unnecessary.